### PR TITLE
[ISSUE-2493] show child nodes path

### PR DESCRIPTION
**Pull request by WilliamSong11**
## Description
Hi all :
        I have submitted a pr for  [ISSUE-2493](https://github.com/apache/iotdb/issues/2493), please take a look.
        This issue about add SQL syntax for metadata queries  

The modules involved are as follows:
&ensp;&ensp;&ensp;1、antlr
&ensp;&ensp;&ensp;2、server/query planner 
&ensp;&ensp;&ensp;3、server/metadata
The local test results are as follows:

image

### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Review by HTHou — COMMENTED** [review]
Well done!
The code seems good to me. 
Would you like to add some user guides?

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 1]
```diff
@@ -1017,6 +1017,18 @@ public MeasurementSchema getSeriesSchema(PartialPath device, String measurement)
     return mtree.getChildNodePathInNextLevel(path);
   }
 
+  /**
+   * Get child node in the next level of the given path.
+   *
+   * <p>e.g., MTree has [root.sg1.d1.s1, root.sg1.d1.s2, root.sg1.d2.s1] given path = root.sg1,
+   * return [d1, d2] given path = root.sg.d1 return [s1,s2]
+   *
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1025]: this add paremater comment..
so @parameter path ....

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java [hunk 2]
```diff
@@ -1170,6 +1170,76 @@ private void findChildNodePathInNextLevel(
...
 
+  /**
+   * Get child node in the next level of the given path.
+   *
+   * <p>e.g., MTree has [root.sg1.d1.s1, root.sg1.d1.s2, root.sg1.d2.s1] given path = root.sg1,
+   * return [d1, d2]
+   *
+   * <p>e.g., MTree has [root.sg1.d1.s1, root.sg1.d1.s2, root.sg1.d2.s1] given path = root.sg1.d1
+   * return [s1, s2]
+   *
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1181]: Same as above

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java [hunk 3]
```diff
@@ -1170,6 +1170,76 @@ private void findChildNodePathInNextLevel(
...
+            res.add(node.getName());
+          } else {
+            findChildNodeInNextLevel(
+                child, nodes, idx + 1, parent + node.getName() + PATH_SEPARATOR, res, length);
+          }
+        }
+      } else if (idx == length) {
+        String nodeName = node.getName();
+        res.add(nodeName);
+      }
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1239]: if node.getChildren().size() == 0 , will....

**Comment by wangchao316** [comment]
could you please add some IT test  in code?,   E2E...
there is same as cli-test...

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java (hunk 3 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1239]: If node.getChildren().size() == 0, the recursion will return. Is there any problem?

Comment (WilliamSong11) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1239]: Thank you for your reply 


 &ensp;I've run some local tests and the results seem to be in line with expectations

&ensp;&ensp;1、Local client testing
IoTDB> SET STORAGE GROUP TO root.gsxgt
SET STORAGE GROUP TO root.gsxgt
Msg: The statement is executed successfully.
IoTDB> show child nodes root.gsxgt
show child nodes root.gsxgt
+-----------+
|child nodes|
+-----------+
+-----------+
Empty set.
It costs 0.005s
IoTDB> CREATE TIMESERIES root.gsxgt.d1 WITH DATATYPE=BOOLEAN, ENCODING=PLAIN
CREATE TIMESERIES root.gsxgt.d1 WITH DATATYPE=BOOLEAN, ENCODING=PLAIN
Msg: The statement is executed successfully.
IoTDB> show child nodes root.gsxgt
show child nodes root.gsxgt
+-----------+
|child nodes|
+-----------+
|         d1|
+-----------+
Total line number = 1
It costs 0.009s
IoTDB> show child nodes root.gsxgt.d1
show child nodes root.gsxgt.d1
+-----------+
|child nodes|
+-----------+
+-----------+
Empty set.
It costs 0.004s
IoTDB> CREATE TIMESERIES root.gsxgt.d1.s1 WITH DATATYPE=BOOLEAN, ENCODING=PLAIN
CREATE TIMESERIES root.gsxgt.d1.s1 WITH DATATYPE=BOOLEAN, ENCODING=PLAIN
Msg: The statement is executed successfully.
IoTDB> show child nodes root.gsxgt.d1
show child nodes root.gsxgt.d1
+-----------+
|child nodes|
+-----------+
|         s1|
+-----------+
Total line number = 1
It costs 0.005s


&ensp;&ensp;2、org.apache.iotdb.db.metadata.MTreeTest --> Add Child Node as an empty scene 
image

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 1 above)
Comment (WilliamSong11) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1025]: Thank you for this advice，I will adjust accordingly

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java (hunk 3 above)
Comment (WilliamSong11) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1239]: Is this the desired effect we want to achieve @wangchao316 @HTHou  @qiaojialin

Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1239]: > Is this the desired effect we want to achieve @wangchao316 @HTHou @qiaojialin

Yes, I think this is the desired result exactly.

**Comment by WilliamSong11** [comment]
@HTHou @wangchao316 
HI all！
&nbsp;I've already made these adjustments:
&nbsp;&nbsp;1、Added documentation about ‘show child nodes’, including Chinese and English
&nbsp;&nbsp;2、Add IT Test  in IoTDBMetadataFetchIT.showChildNodes

File: server/src/main/java/org/apache/iotdb/db/metadata/MTree.java (hunk 3 above)
Comment (wangchao316) [reply server/src/main/java/org/apache/iotdb/db/metadata/MTree.java:1241]: > @HTHou @wangchao316
> HI all！
>  I've already made these adjustments:
>   1、Added documentation about ‘show child nodes’, including Chinese and English
>   2、Added IT Test in IoTDBMetadataFetchIT.showChildNodes

good

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBMetadataFetchIT.java [hunk 4]
```diff
@@ -367,6 +367,41 @@ public void showChildPaths() throws SQLException, ClassNotFoundException {
...
+      for (int n = 0; n < sqls.length; n++) {
+        String sql = sqls[n];
+        String standard = standards[n];
+        StringBuilder builder = new StringBuilder();
+        try {
+          boolean hasResultSet = statement.execute(sql);
+          if (hasResultSet) {
+            try (ResultSet resultSet = statement.getResultSet()) {
+              ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+              while (resultSet.next()) {
```
Comment (wangchao316) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBMetadataFetchIT.java:388]: + 1

**Review by wangchao316 — APPROVED** [review]

**Review by HTHou — APPROVED** [review]

**Pull request merged**